### PR TITLE
server: add DTO beans for /supported-rules response (#429)

### PR DESCRIPTION
**Pull request by innokenty**

### CODE REVIEW

File: server/src/main/java/de/zalando/zally/rule/Rule.kt [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-import com.fasterxml.jackson.annotation.JsonProperty
 import de.zalando.zally.dto.Violation
 import de.zalando.zally.dto.ViolationType
 import io.swagger.models.Swagger
 
 interface Rule {
 
     val title: String
-    @get:JsonProperty("type") val violationType: ViolationType
+    val violationType: ViolationType
```
Comment (netme) [inline server/src/main/java/de/zalando/zally/rule/Rule.kt]: That might be an issue since it's called `type` in API, not `violation_type`

Comment (maxim-tschumak) [inline server/src/main/java/de/zalando/zally/rule/Rule.kt]: Why not just renaming the DOJOs property/fields names?

Comment (netme) [inline server/src/main/java/de/zalando/zally/rule/Rule.kt]: I think with `type` it won't work since it's reserved keyword in Kotlin

**Comment by maxim-tschumak** [comment]
:+1:

File: server/src/main/java/de/zalando/zally/rule/Rule.kt (hunk 1 above)
Comment (innokenty) [inline server/src/main/java/de/zalando/zally/rule/Rule.kt]: That's the point: `Rule` is not used anymore for API reasons. `RuleDTO` serves this purpose now and `Rule` is just an internal interface. Note that `RuleDTO` has this field called just `type`.

**Comment by netme** [comment]
👍

**Pull request merged**